### ISSUE TITLE
Bug: Add support for Typescript 7

### ISSUE THREAD

**Issue by jmtelljohann** [typescript-eslint/typescript-eslint#12720]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Playground Link

NA

### Repro Code

```TypeScript
There doesn't seem to be an appropriate issue type for this. The latest version of typescript-eslint has a peer dependency of typescript 4.8.4 - 6.1.0. I would like to be able to upgrade to typescript 7, but I need typescript-eslint to support it first.
```

### ESLint Config

```javascript
NA
```

### tsconfig

```jsonc
NA
```

### Expected Result

Support for Typescript 7

### Actual Result

Only supports Typescript 4.8.4 - 6.1.0

**Comment by bradzacher** [comment]
Please use search and check pinned issues
https://github.com/typescript-eslint/typescript-eslint/issues/10940
